src/fan.hpp

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2022 YADRO
+
+#pragma once
+
+#include <phosphor-logging/log.hpp>
+
+#include <core/entity/dbus_query.hpp>
+#include <core/entity/entity.hpp>
+#include <status_provider.hpp>
+#include <sensors.hpp>
+
+namespace app
+{
+namespace obmc
+{
+namespace entity
+{
+using namespace app::entity;
+using namespace app::query;
+using namespace phosphor::logging;
+
+class Fan final :
+    public entity::Collection,
+    public CachedSource,
+    public NamedEntity<Fan>
+{
+  public:
+    enum class State
+    {
+        enabled
+    };
+
+    ENTITY_DECL_FIELD(std::string, Id)
+    ENTITY_DECL_FIELD(std::string, Name)
+    ENTITY_DECL_FIELD(std::string, PrettyName)
+    ENTITY_DECL_FIELD(std::string, Type)
+    ENTITY_DECL_FIELD(std::string, Connector)
+
+    ENTITY_DECL_FIELD(std::string, Manufacturer)
+    ENTITY_DECL_FIELD(std::string, Model)
+    ENTITY_DECL_FIELD(std::string, PartNumber)
+    ENTITY_DECL_FIELD(std::string, SerialNumber)
+    
+    ENTITY_DECL_FIELD_DEF(uint8_t, TachCount, 0)
```

Comment:
The REDFISH Thermal schema provides readings for each fan object. It could be Percent or RPM. 
If we have several RPM sensors for a single fan (n110-series), we can't determinate which should be selected to provide that information. If we have exactly one sensor, we will provide RPM sensor reading, if we have two or more sensors, we must provide PWM sensor reading. No other way there is.

Reply:
I'd suggest discussing this in the team chat or a conference call if the explanation is not clear or you have some questions.

Reply:
>>if we have two or more sensors, we must provide PWM sensor reading. No other way there is.
IMHO we can use first (inlet) fan as main one. Otherwise, we can use PWM in all cases.
I guess, same customers could use both 1U and 2U servers and this would be strange for them if one shows speed in RPM and other - in percentage.

Reply:
Your remark makes sense, but I tried to follow the current implementation of REDIFSH (by the bmcweb) in all cases where it is possible, hasn't logical mistakes, and is consistent with DMTF REDFISH schema definition. So, the behavior to select appropriate sensor to provider fan reading is such as I implemented here

Reply:
Summary, I'd prefer to bring up for discussion the right way to provide valid, consistent, and explicit fan reading. 
Your proposal looks good to me; @AlexanderAmelkin, @nest1ing, what do you think about this?

Reply:
Fixed appropriate of conversation